### PR TITLE
CN Stats Page HTML & CSS Layout

### PR DESCRIPTION
**Pull request by chrisn02**
Have made the layout for the user stats page. At the top of the page it says "Stats for [name]" with a graph underneath that shows the percentage of every quiz the user has done, and you can hover over each point in the graph to see the exact percentage. Underneath that is a table that shows data about each individual test. Table may need to some changes but the basic layout is there. It also displays the average score of each test as well as the total number of points earned. There are plans for an export functionality as well, which will be implemented at a later date.
UserStatsPageLayout
UserStatsPageLayout_Hovered

### CODE REVIEW

File: style.css [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+        color: white;
+        margin: 0;
+        padding: 0;
+        box-sizing: border-box;
+    }
+    
+    body
+    {
+        background-color: #181818;
+        margin: 0 auto;
```
Comment (Jforjo) [inline style.css]: Try not to have a margin on the body. You should use things like `display: flex;`, `justify-content: center;` and `flex-direction: center;`.

File: user-stats.php [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    <header>
+        <nav>
+            <ul>
+                <li><a href="#">Home</a></li>
+                <li><a href="#">Stats</a></li>
+                <li><a href="#">Leaderboards</a></li>
+            </ul>
+        </nav>
+    </header>
+    <section class="welcome-section">
```
Comment (Jforjo) [inline user-stats.php:22]: I don't think a section tag should be used here. The `<h2>` inside should wither be within the header, as that would logically make sense, or as it is right now.

File: user-stats.php [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+                        <td>42/50</td>
+                        <td>84%</td>
+                        <td>100</td>
+                    </tr>
+                </tbody>
+            </table>
+        </div>
+    </div>
+
+    <script src="https://cdn.jsdelivr.net/npm/chart.js"></script>
```
Comment (Jforjo) [inline user-stats.php]: Move this into the `<head>` tag at the top and add the defer attribute to it.
It should look like this: `<script defer src="...">`.
This is how all script files should be.

File: user-stats.php [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+
+    <script>
```
Comment (Jforjo) [inline user-stats.php]: You should probably put this within an external script file.

**Review by Jforjo — CHANGES_REQUESTED** [review]
Looks good. Read comments.
__Could__ change the line graph to have it's maximum and minimum values being the highest and lowest scores respectively. This is because if someone keeps scoring between 70% and 80% then it would be hard to see the change, so removing the redundant space above and below would emphasis the users progress more.

**Review by gchodancar19 — APPROVED** [review]
As you are using bootstrap there are some stuff which not needed specially the utility stuff is not necessarily needed.

File: user-stats.php [hunk 5]
```diff
@@ -4,7 +4,9 @@
     <meta charset="UTF-8">
     <meta name="viewport" content="width=device-width, initial-scale=1.0">
     <title>User Stats</title>
-    <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.3.0-alpha3/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-KK94CHFLLe+nY2dmCWGMq91rCGa5gtU4mk92HdvYe+M/SXH301p5ILy+dN9+nJOZ" crossorigin="anonymous">
+    <link defer href="https://cdn.jsdelivr.net/npm/bootstrap@5.3.0-alpha3/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-KK94CHFLLe+nY2dmCWGMq91rCGa5gtU4mk92HdvYe+M/SXH301p5ILy+dN9+nJOZ" crossorigin="anonymous">
```
Comment (Jforjo) [inline user-stats.php]: The defer attribute is specific to script tags, so you can remove it from this link tag.

Comment (chrisn02) [reply user-stats.php]: Didn't mean to add that, thanks for letting me know

**Comment by chrisn02** [comment]
Have made the requested changes. The graph now displays from the user's lowest score to their highest score by changing beginAtZero from true to false, and have also moved the javascript code for the graph into its own file called graph.js
image

**Review by Jforjo — APPROVED** [review]
Front-end looks good.
Although, I would suggest putting all your JavaScript inside an anonymous function, like this:
```javascript
(() => {
    // code...
})();
```
but this isn't necessary.

**Comment by chrisn02** [comment]
Thanks, will look into doing that

**Comment by chrisn02** [comment]
Sorry, resolved a merge conflict so requires review again

**Review by Jforjo — APPROVED** [review]

**Pull request merged**